### PR TITLE
feat: room, story, model hvac sys grouping

### PR DESCRIPTION
**Pull request by TfedUD**
this PR adds to the existing story->hvac system grouping capabilities, the following:
1. one system per model
2. one system per zone

### CODE REVIEW

File: honeybee_doe2/cli/translate.py [hunk 1]
```diff
@@ -19,20 +19,24 @@ def translate():
 @click.argument('hb-json', type=click.Path(
     exists=True, file_okay=True, dir_okay=False, resolve_path=True)
 )
+@click.option('--hvac_grouping', '-hg',
```
Comment (mostaphaRoudsari) [inline honeybee_doe2/cli/translate.py:22]: What if we call it `havc_mapping` instead? My understanding is that this feature maps each room, story, or the whole model to HVAC systems.

File: honeybee_doe2/writer.py [hunk 2]
```diff
@@ -15,8 +15,21 @@
...
+            hvac_grouping: accepts: 'room', 'story', 'model'
+    """
+    hvac_groups = None
+    if hvac_grouping == 'room':
+        hvac_groups = hb_model.properties.doe2.hvac_sys_zones_by_room
+    elif hvac_grouping == 'story':
+        hvac_groups = hb_model.properties.doe2.hvac_sys_zones_by_story
+    elif hvac_grouping == 'model':
+        hvac_groups = hb_model.properties.doe2.hvac_sys_zones_by_model
+
```
Comment (mostaphaRoudsari) [inline honeybee_doe2/writer.py:32]: What if the user uses a wrong input? You need a final check to raise an error for an invalid input.

File: honeybee_doe2/writer.py [hunk 3]
```diff
@@ -15,8 +15,21 @@
 from honeybee.typing import clean_string
 
 
-def model_to_inp(hb_model):
+def model_to_inp(hb_model, hvac_grouping):
```
Comment (mostaphaRoudsari) [inline honeybee_doe2/writer.py:18]: Let's put a default value here for hvac_grouping so it doesn't break the existing codes. It should be set to `story`

File: honeybee_doe2/properties/model.py [hunk 4]
```diff
@@ -96,14 +96,32 @@ def _make_mats_cons_layers(obj):
...
 
+    @property
+    def hvac_sys_zones_by_room(self):
+        return self._get_hvac_sys_zones_by_room(self.host)
+
+    @staticmethod
+    def _get_hvac_sys_zones_by_room(obj):
+        hvac_sys_zones = [HVACSystem.from_room(room) for room in obj.rooms]
+        return hvac_sys_zones
+
```
Comment (mostaphaRoudsari) [inline honeybee_doe2/properties/model.py:124]: Not a big fan of adding many `staticmethod`s. Why don't you just use the property in this case?

```suggestion
    def _get_hvac_sys_by_model(obj):
        hvac_sys = [HVACSystem.from_model(obj)]
        return hvac_sys

    @property
    def hvac_sys_zones_by_story(self):
        hvac_sys_zones = [HVACSystem.from_story(story) for story in self.stories]
        return hvac_sys_zones

    @property
    def hvac_sys_zones_by_room(self):
        hvac_sys_zones = [HVACSystem.from_room(room) for room in self.host.rooms]
        return hvac_sys_zones

```

I would also add some docstrings here for your future self! You will thank yourself for that! 😄

**Review by mostaphaRoudsari — COMMENTED** [review]
Nice work! See my comments. 💪

File: honeybee_doe2/properties/hvac.py [hunk 5]
```diff
@@ -59,19 +60,41 @@ class HVACSystem:
     name: str
     zones: List[Zone]
 
+    @classmethod
```
[the remark this answers is not in the document]
Comment (TfedUD) [reply honeybee_doe2/properties/hvac.py:65]: done! [see commit](https://github.com/ladybug-tools/honeybee-doe2/pull/59/commits/695017329302986c6801849eaa5fe48ad6229785)

File: honeybee_doe2/cli/translate.py (hunk 1 above)
Comment (TfedUD) [reply honeybee_doe2/cli/translate.py]: done! [see commit](https://github.com/ladybug-tools/honeybee-doe2/pull/59/commits/418982bc77511d6e557174a1d372de82fb722fdb)

File: honeybee_doe2/writer.py (hunk 2 above)
Comment (TfedUD) [reply honeybee_doe2/writer.py:39]: done! [see commit](https://github.com/ladybug-tools/honeybee-doe2/pull/59/commits/52b45eb0190036830d0e1645c327ca26d406b2b6)

File: honeybee_doe2/writer.py (hunk 3 above)
Comment (TfedUD) [reply honeybee_doe2/writer.py]: done! [see commit](https://github.com/ladybug-tools/honeybee-doe2/pull/59/commits/f17cf169dad9f6b3757eb7a1e0b47837f59495a1)

File: honeybee_doe2/properties/model.py (hunk 4 above)
Comment (TfedUD) [reply honeybee_doe2/properties/model.py]: done! [see commit](https://github.com/ladybug-tools/honeybee-doe2/pull/59/commits/d5c09e8a92e4ee07d881212f93fd1e19257f584d)

**Pull request merged**